src/main/java/com/example/doodle_clone/controller/MainController.java

```diff
@@ -1,62 +1,87 @@
+
+
 
 import java.util.ArrayList;
+import java.util.List;
 import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
 
 @Controller
 public class MainController {
 
     @Autowired
-    private Repository repository;
+    private MeetingRepository meetingRepository; //репозиторій зустрічі
 
+    @Autowired
+    private SlotRepository slotRepository;
 
 
-    @GetMapping("/main_title")
-    public String mainTitle(Model model){
-        Iterable<Meeting> meetings = repository.findAll();
-        model.addAttribute("meetings",meetings);
+    @GetMapping("/main-title")
+    public String mainTitle(Model model) {
+        Iterable<Meeting> meetings = meetingRepository.findAll();
+        model.addAttribute("meetings", meetings);
         return "mainMenu";
     }
 
-    @GetMapping("/newTable")
-    public String newTable(Model model){
+    @GetMapping("/new-table")
+    public String newTable(Model model) {
         return "newTable";
     }
 
 
-
-    @PostMapping("/newTable")
-    public String createNewTable(@RequestParam String title,@RequestParam String text,@RequestParam String location,@RequestParam int count,Model model){
-        Meeting meetings = new Meeting(title,text,location,count);
-        repository.save(meetings);
-        return "redirect:/main_title";
+    @PostMapping("/new-table")
+    public String createNewTable(@RequestParam String title, @RequestParam String text, @RequestParam String location, @RequestParam int count,
+                                 @RequestParam("name") List<String> slotNames,
+                                 @RequestParam("costMember") List<Integer> slotCosts, Model model) {
+        Meeting meetings = new Meeting(title, text, location, count);
+        meetingRepository.save(meetings);
+        List<Slot> slots = IntStream.range(0, slotNames.size())
+                .mapToObj(i -> {
+                    Slot slot = new Slot(slotNames.get(i), slotCosts.get(i));
+                    slot.setMeeting(meetings);
+                    return slot;
+                })
+                .collect(Collectors.toList());
+        meetings.setSlots(slots);
+        slotRepository.saveAll(slots);
+        return "redirect:/main-title";
     }
 
 
-
-    @GetMapping("/meeting_info/{id}")
-    public String meetingDetails(@PathVariable(value = "id") Long id, Model model){
-        Optional<Meeting> meetings = repository.findById(id);
+    @GetMapping("/meeting-info/{id}")
+    public String meetingDetails(@PathVariable(value = "id") Long id, Model model) {
+        Optional<Meeting> meetings = meetingRepository.findById(id);
         ArrayList<Meeting> res = new ArrayList<>();
         meetings.ifPresent(res::add);
-        model.addAttribute("meetings",res);
+        model.addAttribute("meetings", res);
+
+        List<Slot> slots = slotRepository.findByMeetingId(id);
+        model.addAttribute("slots", slots);
         return "meeting_info";
```

Comment:
why is this returning constant string?

Reply:
I'm not sure if I understood correctly, but our page is returned with information about the table from its ID, the ID is taken from the Java method in the HTML code and displays the information for the table we need, which is taken from the method